### PR TITLE
Add reference inputs (de)serialization + tests

### LINKED ISSUE #680: Vasil features: add support for reference inputs
Follow these changes in CSL: https://github.com/Emurgo/cardano-serialization-lib/pull/461

Modify our (de)serialization and domain types accordingly

### PR DESCRIPTION
**Pull request by klntsky**
Closes #680 

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [ ] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [ ] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)

### CODE REVIEW

File: src/Deserialization/Transaction.js [hunk 1]
```diff
@@ -59,6 +59,9 @@ exports._txBodyAuxiliaryDataHash = maybeGetter("auxiliary_data_hash");
 exports._txBodyValidityStartInterval = maybeGetter("validity_start_interval_bignum");
 // multiassets(): Mint | void;
 exports._txBodyMultiAssets = maybeGetter("multiassets");
+exports._txBodyReferenceInputs = maybe => containerhelper =>
+    body => body.reference_inputs() ?
```
Comment (ngua) [inline src/Deserialization/Transaction.js:63]: I think this formatting is a bit hard to read, but that might just be me

Comment (ngua) [reply src/Deserialization/Transaction.js:63]: Speaking of which we should probably add `prettier` or something and format all of the JS (along with a linter). Unrelated to this PR of course

**Review by ngua — APPROVED** [review]
Small comment about JS formatting but otherwise looks good

File: src/Deserialization/Transaction.js (hunk 1 above)
Comment (klntsky) [reply src/Deserialization/Transaction.js:63]: https://github.com/Plutonomicon/cardano-transaction-lib/issues/530

File: src/Cardano/Types/Transaction.purs [hunk 2]
```diff
@@ -198,6 +198,7 @@ newtype TxBody = TxBody
   , auxiliaryDataHash :: Maybe AuxiliaryDataHash
   , validityStartInterval :: Maybe Slot
   , mint :: Maybe Mint
+  , referenceInputs :: Maybe (Array TransactionInput)
```
Comment (vvtran) [inline src/Cardano/Types/Transaction.purs:201]: Should we use `Set` instead? 
```
 /**
   * @param {TransactionInputs} reference_inputs
   */
  set_reference_inputs(reference_inputs: TransactionInputs): void;
```
There's probably an enforced ordering.

Comment (vvtran) [reply src/Cardano/Types/Transaction.purs:201]: I'm not actually certain on this admittedly.

Comment (klntsky) [reply src/Cardano/Types/Transaction.purs:201]: I'd prefer to preserve the ordering, it's safer to be closer to CTL than not. We don't have `Set` as API requirement, right?

**Pull request merged**